common/raw_hashtable_metadata_group.h

```diff
@@ -0,0 +1,930 @@
+  }
+  *this = UseSIMD ? simd_g : portable_g;
+}
+
+inline auto MetadataGroup::Match(uint8_t present_byte) const -> MatchRange {
+  // The caller should provide us with the present byte hash, and not set any
+  // present bit tag on it so that this layer can manage tagging the high bit of
+  // a present byte.
+  CARBON_DCHECK((present_byte & PresentMask) == 0)
+      << llvm::formatv("{0:x}", present_byte);
+
+  MatchRange portable_result;
+  MatchRange simd_result;
+  if constexpr (!UseSIMD || DebugSIMD) {
+    portable_result = PortableMatch(present_byte);
+  }
+  if constexpr (UseSIMD || DebugSIMD) {
+    simd_result = SIMDMatch(present_byte);
+    CARBON_DCHECK(simd_result == portable_result)
+        << "SIMD result '" << simd_result << "' doesn't match portable result '"
+        << portable_result << "'";
+  }
+  return UseSIMD ? simd_result : portable_result;
+}
+
+inline auto MetadataGroup::MatchPresent() const -> MatchRange {
+  MatchRange portable_result;
+  MatchRange simd_result;
+  if constexpr (!UseSIMD || DebugSIMD) {
+    portable_result = PortableMatchPresent();
+  }
+  if constexpr (UseSIMD || DebugSIMD) {
+    simd_result = SIMDMatchPresent();
+    CARBON_DCHECK(simd_result == portable_result)
+        << "SIMD result '" << simd_result << "' doesn't match portable result '"
+        << portable_result << "'";
+  }
+  return UseSIMD ? simd_result : portable_result;
+}
+
+inline auto MetadataGroup::MatchEmpty() const -> MatchIndex {
+  MatchIndex portable_result;
+  MatchIndex simd_result;
+  if constexpr (!UseSIMD || DebugSIMD) {
+    portable_result = PortableMatchEmpty();
+  }
+  if constexpr (UseSIMD || DebugSIMD) {
+    simd_result = SIMDMatchEmpty();
+    CARBON_DCHECK(simd_result == portable_result)
+        << "SIMD result '" << simd_result << "' doesn't match portable result '"
+        << portable_result << "'";
+  }
+  return UseSIMD ? simd_result : portable_result;
+}
+
+inline auto MetadataGroup::MatchDeleted() const -> MatchIndex {
+  MatchIndex portable_result;
+  MatchIndex simd_result;
+  if constexpr (!UseSIMD || DebugSIMD) {
+    portable_result = PortableMatchDeleted();
+  }
+  if constexpr (UseSIMD || DebugSIMD) {
+    simd_result = SIMDMatchDeleted();
+    CARBON_DCHECK(simd_result == portable_result)
+        << "SIMD result '" << simd_result << "' doesn't match portable result '"
+        << portable_result << "'";
+  }
+  return UseSIMD ? simd_result : portable_result;
+}
+
+inline auto MetadataGroup::CompareEqual(MetadataGroup lhs, MetadataGroup rhs)
+    -> bool {
+  bool portable_result;
+  bool simd_result;
+  if constexpr (!UseSIMD || DebugSIMD) {
+    portable_result = PortableCompareEqual(lhs, rhs);
+  }
+  if constexpr (UseSIMD || DebugSIMD) {
+    simd_result = SIMDCompareEqual(lhs, rhs);
+    CARBON_DCHECK(simd_result == portable_result);
+  }
+  return UseSIMD ? simd_result : portable_result;
+}
+
+inline auto MetadataGroup::VerifyIndexMask(
```

Comment:
I will need to review the Verify* functions again once I know what they are supposed to do.